src/components/TodoList/TodoList.tsx

```diff
@@ -1,12 +1,57 @@
-/* eslint-disable max-len */
-import React from 'react';
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const { query, status } = useAppSelector(state => state.filter);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([...todos]);
+
+  function sortTodos() {
+    let correctListOfTodos;
+
+    switch (status) {
+      case 'active':
+        correctListOfTodos = [...todos].filter(todo => !todo.completed);
+        break;
+
+      case 'completed':
+        correctListOfTodos = [...todos].filter(todo => todo.completed);
+        break;
+
+      default:
+        correctListOfTodos = [...todos];
+        break;
+    }
+
+    setVisibleTodos(correctListOfTodos.filter(todo => (
+      todo.title.toLowerCase()).includes(query.toLowerCase())));
+  }
+
+  useEffect(() => {
+    sortTodos();
+  }, [query, status]);
+
+  function setTargetTodo(id: number) {
+    const targetTodo = todos.find(todo => todo.id === id);
+
+    if (targetTodo) {
+      dispatch(actions.setTodo(targetTodo));
+    }
+  }
+
   return (
     <>
-      <p className="notification is-warning">
-        There are no todos matching current filter criteria
-      </p>
+      {Boolean(!visibleTodos.length) && (
```

Comment:
Actually, you don't need to write `Boolean` here. The logical operator already converts the value to the boolean type
```suggestion
      {!visibleTodos.length && (
```

Reply:
![image](https://user-images.githubusercontent.com/110824803/216968301-af16efb9-2651-41ab-a91b-ea894eb5a939.png)
але він буде відображати 0 =(